include/courtier/GIoContexts.hpp

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include <cstdlib>
+#include <climits>
+#include <thread>
+#include <pthread.h>
+#include <vector>
+#include <mutex>
+
+#include <boost/version.hpp>
+#include <boost/asio.hpp>
+#include <boost/noncopyable.hpp>
+#include <boost/shared_ptr.hpp>
+
+#include "common/GGlobalDefines.hpp"
+#include "common/GLogger.hpp"
+#include "common/GErrorStreamer.hpp"
+
+
+namespace Gem {
+  namespace Courtier {
+
+    namespace net = boost::asio;
+    
+    class GIoContexts
+      : private boost::noncopyable
```

Comment:
boost::noncopyable is actually no longer needed with current C++. You can delete the corresponding copy-, assignment and possibly move constructors and operators.